### ISSUE TITLE
[Patterns] adding more properties to `OSMMesh` workloads

### ISSUE THREAD

**Issue by DelusionalOptimist** [meshery/meshery-osm#110]
**Enhancement Description**
[`OSMMesh`](https://github.com/meshery/meshery-osm/blob/master/templates/oam/workloads/osmmesh_definition.json) workloads, which are used for installing OSM, should have more properties.
This would allow users to have more control over configuration when they install the service mesh.

**Some Considerations**
* Since we're using helm charts for installing OSM, these properties should ideally be extracted from their `Values` file https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml
* We'll need to version these workloads.
* We'll need to figure out how this will keep up with the CLI fallback method.

**Comment by DelusionalOptimist** [comment]
@utkarsh-pro @leecalcote thoughts?

**Comment by leecalcote** [comment]
We're doing away with MeshOps v1 style operations and moving to MeshOps v2, which are pattern-based. Yes, more granular control over the configuration of a service mesh deployment is critical functionality. This functionality needs to be driven through patterns. What aspects of a mesh that are configurable can be, in part, sourced from values.yaml.

**Comment by DelusionalOptimist** [comment]
> Can you confirm that configuration items that are described in values.yaml are not already captured in the pattern components for OSM?

Yeah, I mean we cover a lot of stuff about configuration and I might be missing something :sweat_smile:.
I'll explain from my silly POV.

I tried installing OSM using patterns. I realized I could only specify `version` for OSMMesh workload which is the only field accepted by the [schema](https://github.com/meshery/meshery-osm/blob/master/templates/oam/workloads/osmmesh.meshery.layer5io.schema.json).
```
name: Install OSM
version: 1.0.0
services:
  osm:
    type: OSMMesh
    namespace: osm-system
    settings:
      version: v0.9.1
    traits:
      automaticSidecarInjection.OSM:
        namespaces:
          - default
```
Internally, we'll fetch the helm chart for this version, install it and it would work.

However, we can specify an image tag and a custom container registry for OSM controller's image: https://github.com/openservicemesh/osm/blob/73079be1c0a619f57501785190757c1da9ed2fcc/charts/osm/values.yaml#L9-L15 and I think I might want to use it to get a non-stable release. Internally, we could just install by giving this configuration as `Values` while applying the helm chart.
Idk if this can be specified somehow by applying a CRD. So not sure if it can be done using the current  workloads, which are only based on CRDs till now.
